pkg/proc/core/linux_core.go

```diff
@@ -129,8 +198,8 @@ type Note struct {
 	Desc interface{} // Decoded Desc from the
 }
 
-// readNotes reads all the notes from the notes prog in core.
-func readNotes(core *elf.File) ([]*Note, error) {
+// readNotesAMD64 reads all the notes from the notes prog in core.
+func readNotesAMD64(core *elf.File) ([]*Note, error) {
```

Comment:
This function and `readNotesARM64` are basically the same, remove `readNotesARM64` call this one `readNotes` and add `machineType` as a parameter.